lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs, isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
+
   let npmTarget;
   let npmArgs;
   let result;
   switch (yarnTarget) {
     case "install":
       npmTarget = "install";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
       break;
     case "add":
       npmTarget = "install";
       const packages = yarnSubCommands.map(transformNarnPackageString);
-      const saveType = yarnArgs.dev || yarnArgs.D ? "--save-dev" : "--save";
+      const saveType = yarnArgs.dev ? "--save-dev" : "--save";
       npmArgs = [saveType, ...packages];
       break;
     case "remove":
       npmTarget = "uninstall";
-      npmArgs = yarnSubCommands;
+      npmArgs = narnSubArgs;
       break;
     case "init":
       npmTarget = "init";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
+      break;
+    case "update":
+    case "upgrade":
+      npmTarget = "update";
+      npmArgs = narnSubArgs;
+      break;
+    case "list":
+    case "prune":
+    case "version":
+      // commands that don't need further changes
+      npmTarget = yarnTarget;
+      npmArgs = narnSubArgs;
       break;
     case "upgrade-interactive":
-      result = ["-iu", "&&", "npm", "install"];
-      result.command = "ncu";
+      if (isPnpm) {
+        result = ["-i", ...narnSubArgs];
```

Comment:
Does pnpm support `pnpm upgrade-interactive --latest`? What other flags does it support?

Reply:
Yes, it does, I already mentioned it before in another comment tho.

[here it is](https://pnpm.js.org/en/cli/update#interactive-i)

Reply:
Another thing to keep in mind tho, is the fact that when you are doing "narn upgrade-interactive" you are assuming the transformation to npm being "yarn upgrade-interactive --latest", right now on the pnpm version I would have to add the "--latest" argument manually to keep that behaviour